### PR TITLE
countcross patch

### PR DESCRIPTION
**Pull request by SansPapyrus683**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

File: solutions/usaco-716.mdx [hunk 1]
```diff
@@ -2,84 +2,262 @@
...
-const int MX = 1e2;
+struct Road {
+	int sr, sc;
+	int er, ec;
+};
 
-bool vis[MX][MX];
-int cows[MX][MX], di[4] = {1, -1, 0, 0}, dj[4] = {0, 0, 1, -1}, nCows, n, k, r;
-set<tuple<int, int, int, int>> roads;
+inline bool operator<(const Road& r1, const Road& r2) {
```
Comment (envyaims) [inline solutions/usaco-716.mdx:26]: some comments about what is going on here would be nice

File: solutions/usaco-716.mdx [hunk 2]
```diff
@@ -2,84 +2,262 @@
...
+	vector<vector<bool>> has_cow(side_len, vector<bool>(side_len));
+	for (int cow = 0; cow < cow_num; cow++) {
+		int r, c;
+		cin >> r >> c;
+		has_cow[--r][--c] = true;
+	}
 
-	cin >> n >> k >> r;
+	vector<vector<bool>> visited(side_len, vector<bool>(side_len));
+	function<int(int, int, int, int)> connected_cow_num;
```
Comment (envyaims) [inline solutions/usaco-716.mdx:60]: comments!!

**Review by ryanchou-dev — COMMENTED** [review]
pls add comments

**Review by ryanchou-dev — APPROVED** [review]
image

**Review by envyaims — APPROVED** [review]
spicy

**Pull request merged**